packages/playground/src/pages/repo-settings-rules-page.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useState } from 'react'
+import {
+  Button,
+  ButtonGroup,
+  Icon,
+  StackedList,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Text,
+  Textarea,
+  Checkbox
+} from '@harnessio/canary'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+import { FormFieldSet } from '../index'
+import { MessageTheme } from '../components/form-field-set'
+
+const formSchema = z.object({
+  name: z.string().min(1, 'Name is required'),
+  description: z.string().min(1, 'Description is required'),
+  targetPatterns: z.string().min(1, 'Patterns are required'),
+  includePatterns: z.enum(['', '1', '2', '3']),
+  toggleValue: z.boolean(),
+  bypassValue: z.string().optional(),
+  access: z.enum(['1', '2'], {}),
+  defaultBranchValue: z.boolean().optional(),
+  editPermissionsValue: z.boolean().optional(),
+  rules: z.array(
+    z.object({
+      id: z.string(),
+      checked: z.boolean()
+    })
+  )
+})
+export type FormFields = z.infer<typeof formSchema>
+// interface RepoSettingsRulesPageProps {
+//   onFormSubmit?: (data: FormFields) => void
+//   onFormCancel?: () => void
+//   apiError?: string | null
+//   isLoading?: boolean
+//   isSuccess?: boolean
+// }
+const rules = [
+  {
+    id: 'request-approval',
+    label: 'Request approval of new changes',
+    description: 'Require re-approval when there are new changes in the pull request'
+  },
+  {
+    id: 'change-requests',
+    label: 'Require resolution of change requests',
+    description: 'All change requests on a pull request must be resolved before it can be merged'
+  },
+  {
+    id: 'comment-resolution',
+    label: 'Require comment resolution',
+    description: 'All comments on a pull request must be resolved before it can be merged'
+  },
+  {
+    id: 'status-checks',
+    label: 'Require status checks to pass',
+    description: 'Selected status checks must pass before a pull request can be merged'
+  },
+  {
+    id: 'merge-strategies',
+    label: 'Limit merge strategies',
+    description: 'Limit which merge strategies are available to merge a pull request'
```

Comment:
this rule is more than a boolean

Reply:
![image](https://github.com/user-attachments/assets/da321941-6652-4ab0-83b0-f0fe534e2618)